database/gormdb_course.go

```diff
@@ -17,6 +17,9 @@ func (db *GormDB) CreateCourse(courseCreatorID uint64, course *pb.Course) error
 	var courses int64
 	if err := db.conn.Model(&pb.Course{}).Where(&pb.Course{
 		OrganizationID: course.OrganizationID,
+	}).Or(&pb.Course{
```

Comment:
We need to document what this does. Suggest to add a comment to the godoc line, e.g., something like:
```go
// CreateCourse creates a new course if user with given ID is admin, enrolls user as course teacher.
// The provided course must have a unique (GitHub) OrganizationID not already associated with existing course.
// Similarly, the course must have a unique course code and year.
```
Please tell me if this is not accurate; feel free to improve it.